main/php_network.h

```diff
@@ -204,18 +204,34 @@ static inline int php_pollfd_for_ms(php_socket_t fd, int events, int timeout)
 /* emit warning and suggestion for unsafe select(2) usage */
 PHPAPI void _php_emit_fd_setsize_warning(int max_fd);
 
+static inline bool _php_check_fd_setsize(int max_fd, int setsize)
+{
+#ifdef PHP_WIN32
+	if (setsize + 1 >= FD_SETSIZE) {
+		_php_emit_fd_setsize_warning(setsize);
+		return false;
+	}
+#else
+	if (max_fd >= FD_SETSIZE) {
+		_php_emit_fd_setsize_warning(max_fd);
+		return false;
+	}
+#endif
+	return true;
+}
+
 #ifdef PHP_WIN32
 /* it is safe to FD_SET too many fd's under win32; the macro will simply ignore
  * descriptors that go beyond the default FD_SETSIZE */
 # define PHP_SAFE_FD_SET(fd, set)	FD_SET(fd, set)
 # define PHP_SAFE_FD_CLR(fd, set)	FD_CLR(fd, set)
 # define PHP_SAFE_FD_ISSET(fd, set)	FD_ISSET(fd, set)
-# define PHP_SAFE_MAX_FD(m, n)		do { if (n + 1 >= FD_SETSIZE) { _php_emit_fd_setsize_warning(n); }} while(0)
+# define PHP_SAFE_MAX_FD(m, n)		_php_check_fd_setsize(m, n)
 #else
 # define PHP_SAFE_FD_SET(fd, set)	do { if (fd < FD_SETSIZE) FD_SET(fd, set); } while(0)
 # define PHP_SAFE_FD_CLR(fd, set)	do { if (fd < FD_SETSIZE) FD_CLR(fd, set); } while(0)
 # define PHP_SAFE_FD_ISSET(fd, set)	((fd < FD_SETSIZE) && FD_ISSET(fd, set))
-# define PHP_SAFE_MAX_FD(m, n)		do { if (m >= FD_SETSIZE) { _php_emit_fd_setsize_warning(m); m = FD_SETSIZE - 1; }} while(0)
+# define PHP_SAFE_MAX_FD(m, n)		_php_check_fd_setsize(m, n)
```

Comment:
This macro previously modified `m`, this should be documented in `UPGRADING.internals` or replicated by passing a reference to `m`.

Reply:
Good catch, thank you